src/test/xmeta/test_symbol_table.cpp

```diff
@@ -517,3 +515,377 @@ TEST_CASE("Resolving types across namespaces test")
+    auto const& it = namespaces.find("N");
+    REQUIRE(it != namespaces.end());
+    auto const& ns_bodies = it->second->get_namespace_bodies();
+    REQUIRE(ns_bodies.size() == 1);
+    auto const& interfaces = ns_bodies[0]->get_interfaces();
+    REQUIRE(interfaces.size() == 1);
+
+    REQUIRE(interfaces.find("IControl") != interfaces.end());
+    auto const& model = interfaces.at("IControl");
+
+    auto const& properties = model->get_properties();
+    REQUIRE(properties[0]->get_id() == "property1");
+
+    auto const& property_type = properties[0]->get_type().get_semantic();
+    REQUIRE((property_type.is_resolved() && std::get<simple_type>(property_type.get_resolved_target()) == simple_type::Int32));
+
+    auto const& methods = model->get_methods();
+    REQUIRE(methods.size() == 2);
+
+    auto const& method0 = methods[0];
+    REQUIRE(method0->get_id() == "get_property1");
+    auto method0_return_type = method0->get_return_type()->get_semantic();
+    REQUIRE((method0_return_type.is_resolved() && std::get<simple_type>(method0_return_type.get_resolved_target()) == simple_type::Int32));
+
+    auto const& method1 = methods[1];
+    REQUIRE(method1->get_id() == "set_property1");
+    REQUIRE(!method1->get_return_type());
+}
+
+TEST_CASE("Resolving Interface property type ref test")
+{
+    std::istringstream test_idl{ R"(
+        namespace N
+        {
+            struct S1
+            {
+            };
+
+            interface IControl
+            {
+                S1 property1;
+                M.S2 property2;
+            }
+        }
+        namespace M
+        {
+            struct S2
+            {
+            };
+        }
+    )" };
+
+    xmeta_idl_reader reader{ "" };
+    reader.read(test_idl);
+    REQUIRE(reader.get_num_syntax_errors() == 0);
+
+    auto const& namespaces = reader.get_namespaces();
+    auto const& it = namespaces.find("N");
+    REQUIRE(it != namespaces.end());
+    auto const& ns_bodies = it->second->get_namespace_bodies();
+    REQUIRE(ns_bodies.size() == 1);
+    auto const& interfaces = ns_bodies[0]->get_interfaces();
+    REQUIRE(interfaces.size() == 1);
+
+    REQUIRE(interfaces.find("IControl") != interfaces.end());
+    auto const& model = interfaces.at("IControl");
+
+    auto const& properties = model->get_properties();
+    {
+        REQUIRE(properties[0]->get_id() == "property1");
+        auto const& property_type = properties[0]->get_type().get_semantic();
+        REQUIRE(property_type.is_resolved());
+        REQUIRE(std::get<std::shared_ptr<struct_model>>(property_type.get_resolved_target())->get_id() == "S1");
+    }
+    {
+        REQUIRE(properties[1]->get_id() == "property2");
+        auto const& property_type = properties[1]->get_type().get_semantic();
+        REQUIRE(property_type.is_resolved());
+        REQUIRE(std::get<std::shared_ptr<struct_model>>(property_type.get_resolved_target())->get_id() == "S2");
+    }
+}
+
+TEST_CASE("Interface event test")
```

Comment:
Similar to the property tests, seems like we should have tests validating that the event is properly linked to its add/remove methods.